File: OCH_Tao/ABC/B/197.py
H,W,X,Y = map(int,input().split())
S = []
for i in range(H):
  I = list(input())
  if i!=X-1:
    S.append(I)
  else:
    I[Y-1]="X"
    S.append(I)
L = []
L+=("".join(S[X-1]).split("#"))
L+=("".join([i[Y-1] for i in S]).split("#"))
print(len("".join([i for i in L if "X" in i]))-1)

File: OCH_Tao/ABC/B/197.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [H, W, X, Y] = input.shift().split(" ").map(x => parseInt(x, 10));
let cnt = -3;
for (let i = X - 1; i < H; i++) {
  if (input[i][Y - 1] === "#") {
    break;
  } else {
    cnt++;
  }
}
for (let i = X - 1; i >= 0; i--) {
  if (input[i][Y - 1] === "#") {
    break;
  } else {
    cnt++;
  }
}
for (let i = Y - 1; i < W; i++) {
  if (input[X - 1][i] === "#") {
    break;
  } else {
    cnt++;
  }
}
for (let i = Y - 1; i >= 0; i--) {
  if (input[X - 1][i] === "#") {
    break;
  } else {
    cnt++;
  }
}
console.log(cnt);